en/micro-integrator/docs/setup/security/encrypting_plain_text.md

```diff
@@ -1,118 +1,138 @@
+1.  [Download](https://wso2.com/integration/micro-integrator/tooling/) and setup the Micro Integrator CLI tool.
+2.  Initialize the CLI tool from your command line:
 
-### Enter password in command-line
-1. Start the server by running the product start up script from the MI_HOME/bin/ directory as shown below.
-   ```bash
-   ./micro-integrator.sh
-   ```
-2. When you run the startup script, the following message will be prompted before starting the server: `[Enter KeyStore and Private Key Password:]`. This is because, in order to connect to the default user store, the encrypted passwords should be decrypted. The administrator starting the server must provide the private key and keystore passwords using the command-line. Note that passwords are hidden from the terminal and log files.
+    ```bash
+    ./mi
+    ```
+3.  Initialize the secret creation process in the tool:
+
+    ```bash
+    mi secret init
+    ```
+
+    You will be prompted to provide details of the keystore that will be used for encryption.
 
-### Start server as a background job
+    - Keystore location
+    - keystore type
+    - keystore alias
+    - Keystore password
 
-If you start the Micro Integrator as a background job, you will not be able to provide password values on the command line. Therefore, you must start the server in "daemon" mode as explained below.
+4.  Execute one of the following commands to generate the secret:
 
-1. Create a new file in the MI_HOME directory. The file should be named according to your OS as explained below.
+    ```bash
+    # To encrypt secret and get output to console
+    mi secret create
 
-    * For Linux: The file name should be `password-tmp`.
-    * For Windows: The file name should be `password-tmp.txt`.
+    # To encrypt secret and get output to file
+    mi secret create file
+
+    # To encrypt secret and get output as a .yaml file
+    mi secret create k8
+
+    # To bulk encrypt secrets defined in a properties file
+    mi secret create -f=</file_path>
+    ```
 
-    !!! Note
-        When you start the server, the keystore password will be picked from this new file. Note that this file is automatically deleted from the file system after the server starts. Therefore, the admin has to create a new text file every time the server starts.
+### Using encrypted (dynamic) secrets
 
-        Alternatively, if you want to retain the password file after the server starts, the file should be named as follows:
+To dynamically load secrets to server configurations, update the parameters in the `deployment.toml` file as an environment variable or system property:
 
-        * For Linux: The file name should be `password-persist`
-        * For Windows: The file name should be `password-persist.txt`
+```toml tab='Environment Variable'
+[keystore.primary]
+password = "$env{ENV_VAR}"
+alias = "$env{ENV_VAR}"
+key_password = "$$env{ENV_VAR}"  
```

Comment:
$$ small error in env variable definition

Reply:
Fixed